### PR TITLE
Add plugin: Better Checklist

### PR DESCRIPTION
**Pull request by chardskarth**
# I am submitting a new Community Plugin

## Repo URL

Link to my plugin: https://github.com/chardskarth/obsidian-better-checklist

## Release Checklist
- [x] I have tested the plugin on
  - [ ]  Windows
  - [x]  macOS
  - [ ]  Linux
  - [x]  Android _(if applicable)_
  - [x]  iOS _(if applicable)_
- [x] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

### CODE REVIEW

**Comment by joethei** [comment]
[# Obsidian Better Checklist](https://github.com/chardskarth/obsidian-better-checklist/blob/b5e33baa2a8a8a2140e55e23bcffd5cb885ed2f6/README.md#L4)
Naming something "Obsidian xyz" is reserved for first party products we create, please change this header.

[name: 'Show Checklist Pane',](https://github.com/chardskarth/obsidian-better-checklist/blob/b5e33baa2a8a8a2140e55e23bcffd5cb885ed2f6/src/main.ts#L22), [name: 'Reload Checklist',](https://github.com/chardskarth/obsidian-better-checklist/blob/b5e33baa2a8a8a2140e55e23bcffd5cb885ed2f6/src/main.ts#L56), [.setName('Look and Feel')](https://github.com/chardskarth/obsidian-better-checklist/blob/b5e33baa2a8a8a2140e55e23bcffd5cb885ed2f6/src/settings.ts#L66), [.setName('Auto Refresh List?')](https://github.com/chardskarth/obsidian-better-checklist/blob/b5e33baa2a8a8a2140e55e23bcffd5cb885ed2f6/src/settings.ts#L77), [thisSetting.setName('Create a Filter')](https://github.com/chardskarth/obsidian-better-checklist/blob/b5e33baa2a8a8a2140e55e23bcffd5cb885ed2f6/src/settings.ts#L94), [.setButtonText("Add Filter")](https://github.com/chardskarth/obsidian-better-checklist/blob/b5e33baa2a8a8a2140e55e23bcffd5cb885ed2f6/src/settings.ts#L105), [.setName('Filter Name')](https://github.com/chardskarth/obsidian-better-checklist/blob/b5e33baa2a8a8a2140e55e23bcffd5cb885ed2f6/src/settings.ts#L247), [.setName('Include Files')](https://github.com/chardskarth/obsidian-better-checklist/blob/b5e33baa2a8a8a2140e55e23bcffd5cb885ed2f6/src/settings.ts#L253), [.setName("Limit Todos")](https://github.com/chardskarth/obsidian-better-checklist/blob/b5e33baa2a8a8a2140e55e23bcffd5cb885ed2f6/src/settings.ts#L276), ['Group By'](https://github.com/chardskarth/obsidian-better-checklist/blob/b5e33baa2a8a8a2140e55e23bcffd5cb885ed2f6/src/settings.ts#L285), [text: "Checklist Filters",](https://github.com/chardskarth/obsidian-better-checklist/blob/b5e33baa2a8a8a2140e55e23bcffd5cb885ed2f6/src/settings.ts#L425)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

[workspace.setActiveLeaf(todoLeaf, true, true)](https://github.com/chardskarth/obsidian-better-checklist/blob/b5e33baa2a8a8a2140e55e23bcffd5cb885ed2f6/src/main.ts#L42), [workspace.setActiveLeaf(views[0], true, true)](https://github.com/chardskarth/obsidian-better-checklist/blob/b5e33baa2a8a8a2140e55e23bcffd5cb885ed2f6/src/main.ts#L50)
This way of calling `setActiveLeaf` is deprecated, use the other form with the `params` parameter.

[this.app.workspace.getLeavesOfType(TODO_VIEW_TYPE)[0]?.detach()](https://github.com/chardskarth/obsidian-better-checklist/blob/b5e33baa2a8a8a2140e55e23bcffd5cb885ed2f6/src/main.ts#L86)
Do not detach leaves with your custom view in `onunload`, this is an antipattern, see: <https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Don't+detach+leaves+in+%60onunload%60>

[this.containerEl.createEl('h3', {](https://github.com/chardskarth/obsidian-better-checklist/blob/b5e33baa2a8a8a2140e55e23bcffd5cb885ed2f6/src/settings.ts#L48-L50), [this.containerEl.createEl("h4", {](https://github.com/chardskarth/obsidian-better-checklist/blob/b5e33baa2a8a8a2140e55e23bcffd5cb885ed2f6/src/settings.ts#L61-L63)
[Don't add a top-level heading in the settings tab, such as "General", "Settings", or the name of your plugin.](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Only+use+headings+under+settings+if+you+have+more+than+one+section.)

[.setName('Auto Refresh List?')](https://github.com/chardskarth/obsidian-better-checklist/blob/b5e33baa2a8a8a2140e55e23bcffd5cb885ed2f6/src/settings.ts#L77)
There should not be a question mark in this text.

[this.containerEl.createEl("h4", {](https://github.com/chardskarth/obsidian-better-checklist/blob/b5e33baa2a8a8a2140e55e23bcffd5cb885ed2f6/src/settings.ts#L424-L428)
For section headings in settings use:
```ts
new Setting(containerEl).setName('name here').setHeading();
```

[export const isMacOS = () => window.navigator.userAgent.includes('Macintosh')](https://github.com/chardskarth/obsidian-better-checklist/blob/b5e33baa2a8a8a2140e55e23bcffd5cb885ed2f6/src/utils/helpers.ts#L6)
Use `Platform.isMacOS` instead.

[contentDiv.insertAdjacentHTML('beforeend', item.rawHTML);](https://github.com/chardskarth/obsidian-better-checklist/blob/b5e33baa2a8a8a2140e55e23bcffd5cb885ed2f6/src/svelte/ChecklistItem.svelte#L34)
[Avoid using `insertAdjacentHTML`](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Avoid+%60innerHTML%60%2C+%60outerHTML%60+and+%60insertAdjacentHTML%60)

**Comment by chardskarth** [comment]
Hi @joethei,
Thanks for reviewing. I've addressed these except the last one. 

What alternatives do you suggest if I aim to directly add contents from the markdown files? Thanks!

**Comment by joethei** [comment]
See the docs that are linked,

**Comment by chardskarth** [comment]
hi @joethei 

https://github.com/chardskarth/obsidian-better-checklist/commit/f169989c844995866d44303b4f99a62de6a54e2f

Sorry, it maybe just me but I really dont see a clear alternative for my usecase. I found `MarkdownRenderer.render()` but I'm not sure how to specify the other 2 parameters, `sourcePath` and `component`. This commit is working but it shows an initial error:

image

**Comment by chardskarth** [comment]
Ooh, I found [this](https://forum.obsidian.md/t/how-to-access-complete-renderer-text-of-current-markdown-view-at-least-dataviewjs-blocks-and-inline-code/88686)

Revised it in my latest commit. Thanks @joethei for reviewing again.

**Comment by joethei** [comment]
[MarkdownRenderer.render(app, tagStripped, contentDiv, file.file.path, app.plugins.plugins['better-checklist'])](https://github.com/chardskarth/obsidian-better-checklist/blob/49517899fb64b4270b98172a7ead07ce090f8086/src/utils/tasks.ts#L211)
https://github.com/chardskarth/obsidian-better-checklist/blob/49517899fb64b4270b98172a7ead07ce090f8086/src/utils/tasks.ts#L211

Also, the component (last parameter), should be much shorter lived than the plugin instance, currently this can lead to a memory leak because it is never cleaned up in the lifetime of this plugin.

**Comment by chardskarth** [comment]
Does this work @joethei 
https://github.com/chardskarth/obsidian-better-checklist/commit/958f0c4022b4155b59b8c4a754b9ab4488bd3df9

Kindly let me know, thanks.

**Comment by joethei** [comment]
You also need to unload the component once you are no longer using it.
